src/common/file_location.c

```diff
@@ -137,58 +157,122 @@ char *dt_loc_find_install_dir(const char *suffix, const char *searchname)
 
 int dt_loc_init_tmp_dir(const char *tmpdir)
 {
-  darktable.tmpdir = dt_loc_init_generic(tmpdir, g_get_tmp_dir());
+  darktable.tmpdir = dt_loc_init_generic(tmpdir, NULL, g_get_tmp_dir());
+  dt_check_opendir("darktable.tmpdir", darktable.tmpdir, FALSE);
   if(darktable.tmpdir == NULL) return 1;
   return 0;
 }
 
 void dt_loc_init_user_cache_dir(const char *cachedir)
 {
   char *default_cache_dir = g_build_filename(g_get_user_cache_dir(), "darktable", NULL);
-  darktable.cachedir = dt_loc_init_generic(cachedir, default_cache_dir);
+  darktable.cachedir = dt_loc_init_generic(cachedir, NULL, default_cache_dir);
+  dt_check_opendir("darktable.cachedir", darktable.cachedir, TRUE);
   g_free(default_cache_dir);
 }
 
-void dt_loc_init_plugindir(const char *plugindir)
+void dt_loc_init_plugindir(const char* application_directory, const char *plugindir)
 {
 #if defined(__APPLE__) || defined(_WIN32)
   char *suffix = g_build_filename("lib", "darktable", NULL);
   char *directory = dt_loc_find_install_dir(suffix, darktable.progname);
   g_free(suffix);
-  darktable.plugindir = dt_loc_init_generic(plugindir, directory ? directory : DARKTABLE_LIBDIR);
+  darktable.plugindir = dt_loc_init_generic(plugindir, application_directory, directory ? directory : DARKTABLE_LIBDIR);
+  dt_check_opendir("darktable.plugindir", darktable.plugindir, TRUE);
   g_free(directory);
 #else
-  darktable.plugindir = dt_loc_init_generic(plugindir, DARKTABLE_LIBDIR);
+  darktable.plugindir =  dt_loc_init_generic(plugindir, application_directory, DARKTABLE_LIBDIR);
+  dt_check_opendir("darktable.plugindir", darktable.plugindir, TRUE);
 #endif
 }
 
-void dt_loc_init_localedir(const char *localedir)
+void dt_check_opendir(const char* text, const char* directory, gboolean exit_on_error)
+{
+  if (!directory) {
+    printf("directory for %s has not been set.\n", text);
+    exit(EXIT_FAILURE);
+  } 
+
+  DIR* dir = opendir(directory);
+  if (dir) {
+    dt_print(DT_DEBUG_DEV, "%s: %s\n", text, directory);
+    closedir(dir);
+  } 
+  else 
+  {
+    if ( errno == EACCES ) 
+    {
+      printf("Permission denied.\n");
+    } 
+    else if ( errno == ENOENT ) 
+    {
+      printf("%s %s does not exist.\n", text, directory);
+    } 
+    else if ( errno == EBADF ) 
+    {
+      printf("fd is not a valid file descriptor opened for reading.\n");
```

Comment:
But wait aren't all those strings already available and translated when using strerror() ?

Reply:
@TurboGit I wasnt aware of strerror(). I can remove all my messages and replace them with strerror() if thats ok for you. The less code the better. I just dont quite understand `And message should not use capital-letter`, does that mean I have to lowercase every character in my messages or just not the first character of the message?

Reply:
what about "NAME_MAX" and "PATH_MAX"?

Reply:
Should be kept in capital letters. But I suspect all this will be remove if using strerror()

Reply:
> does that mean I have to lowercase every character in my messages or just not the first character of the message

Yes, look at the current code, all strings are lowercase. This is controversial and would be in favor of changing it, but that's the style at the moment and we want to keep that consistent.